### PR TITLE
Move exceptions to specific folder

### PR DESCRIPTION
**Pull request by joubertredrat**
This is only a suggestion, concentrate all Exceptions into `Exception` folder.

Example can be viewed in https://github.com/joubertredrat/queue-interop/tree/master/src

Maybe related to #4

### CODE REVIEW

**Comment by pinepain** [comment]
How do you plan to address broken API BC?

**Comment by joubertredrat** [comment]
Hi @pinepain 

As I viewed on @makasim blog, you have a plan to release 1.0 version in 2018 December and alpha in 2018 March. 
How is a major version, based on SemVer, don't have problem if don't have BC support.

This change is only to make files and folder more readable. When I looked for first time, it seemed a bit confusing.

**Comment by pinepain** [comment]
@joubertredrat Oh, I see, thanks for clarifying you point.

It looks like the benefit of this change are largely void by BC-break and while major version supposed to be released soon, this excuse is not good enough to break BC so close to the major release date without having something significant to bring up.

While you can bump major version and add as many breaks as you want, it normally doesn't help library adoption, especially if it's supposed to provide interoperability. All 0.x changes were a preparation for major release and we passed quite intensive way full of challenges to get here. Unless there is something which has not been shown yet or we missed something, I would propose to delay adding such change which someone may find subjective. 

TL;DR

To avoid potential devastating impact on end-users which BC break normally brings, I would propose to come back to this question later.
